pkg/sqlite/stranded_test.go

```diff
@@ -0,0 +1,76 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStrandedBundleRemover(t *testing.T) {
+	logrus.SetLevel(logrus.DebugLevel)
+	db, cleanup := CreateTestDb(t)
+	defer cleanup()
+	store, err := NewSQLLiteLoader(db)
+	require.NoError(t, err)
+	require.NoError(t, store.Migrate(context.TODO()))
+
+	query := NewSQLLiteQuerierFromDb(db)
+
+	graphLoader, err := NewSQLGraphLoaderFromDB(db)
+	require.NoError(t, err)
+
+	populate := func(name string) error {
+		return registry.NewDirectoryPopulator(
+			store,
+			graphLoader,
+			query,
+			map[image.Reference]string{
+				image.SimpleReference("quay.io/test/" + name): "./testdata/strandedbundles/" + name,
+			}).Populate(registry.ReplacesMode)
+	}
+	for _, name := range []string{"prometheus.0.14.0", "prometheus.0.15.0"} {
+		require.NoError(t, populate(name))
+	}
+
+	// check that the bundle is orphaned
+	querier := NewSQLLiteQuerierFromDb(db)
+	packageBundles, err := querier.GetBundlesForPackage(context.TODO(), "prometheus")
+	require.NoError(t, err)
+	require.Equal(t, 1, len(packageBundles))
+
+	rows, err := db.QueryContext(context.TODO(), "select * from operatorbundle")
+	require.NoError(t, err)
+	require.Equal(t, 2, rowCount(rows))
+	require.NoError(t, rows.Close())
+
+	// prune the orphaned bundle
+	removedBundles, err := store.RemoveStrandedBundles()
```

Comment:
I think this should test the case where multiple bundles are stranded since that's a big part of the pruning